wordpress-mobile/WordPress-Android
# [Site Name] Adapt the Site name screen layout for landscape orientation

Fixes #16294

The Site Name screen layout for landscape orientation is adapted:
- Title text updated to 'Site name' (instead of `Site topic`)
- The title and subtitle in header is hidden, similar to what we do in the other site creation screens.
- Continue button is only shown in portrait orientation when the input is not empty.

I went with 'Site name' instead of the suggested 'Site **Name**' title because during testing it seemed unpolished to see 'Site intent', 'Site Name' and then 'Choose a design' in the site creation flow.

Ideally we'd capitalize the title also for the 'Site intent' screen if we prefer Capitalized Words notation.

## To test:
Prerequisites
<details>
<summary>Toggle the Site Intent Question feature</summary>

1. From My Site Go to Me → App Settings → Debug settings
2. Scroll to Features in development section and tap on SiteIntentQuestionFeatureConfig
3. Tap RESTART THE APP button
4. Since this feature is A/B tested the Site Intent Question will appear only to half of the users. Retry with a different account if it doesn't work

**Alternatively hardcode the feature on by returning `true` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/521e6b3a296339bec026f8ee2648eab326d31129/WordPress/src/main/java/org/wordpress/android/util/config/SiteIntentQuestionFeatureConfig.kt#L19)** 
</details>
<details>
<summary>Toggle the Site Name feature</summary>

1. From My Site Go to Me → App Settings → Debug settings
2. Scroll to Features in development section and tap on SiteNameFeatureConfig
3. Tap RESTART THE APP button
4. Since this feature is A/B tested go to **Abacus**, find the `wpandroid_site_name_v1` experiment and assign to your test a8c-user the treatment variation

**Alternatively hardcode the feature on by returning `true` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/521e6b3a296339bec026f8ee2648eab326d31129/WordPress/src/main/java/org/wordpress/android/util/config/SiteNameFeatureConfig.kt#L19)** 
</details>

1. Toggle the Site Name feature ON
2. Toggle the Site Intent Question feature ON
3. Start the Site Creation flow
4. Choose an intent for your site
5. `Verify` that the Site Name screen is shown
6. Rotate the device or simulator
7. `Verify` that the tile is 'Site name' and the input is the only thing visible besides the top bar.
8. Type something in the input
9. `Verify` that the continue button is not shown

Further testing — Opening the screen in **landscape** device orientation

10. Navigate back
11. Choose an intent for your site
12. `Verify` by repeating steps 7, 8, 9.

## Preview
| Before | After |
| --- | --- |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/163078049-5db21590-a15b-46ad-9ace-1d3dde285c00.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/163076842-4d03092b-8273-4312-bbcd-49de7630d688.png"> |

## Other Changes
- I reused the layout for the header (title & subtitle) that we have on the Site Intent screen, and renamed it to `site_creation_header_v2`.

## Regression Notes
1. Potential unintended areas of impact
   Site Intent screen because I refactored the layout of the title & subtitle to be reusable — though unlikely given that the changes are trivial.

13. What I did to test those areas of impact (or what existing automated tests I relied on)
   Tested manually with different exploratory scenarios.

14. What automated tests I added (or what prevented me from doing so)
   The PR only does UI fixes and UI refactoring → no tests were added or changed.

PR submission checklist:
